src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -2729,6 +2729,23 @@ public GHContent getReadme() throws IOException {
         return requester.withUrlPath(getApiTailUrl("readme")).fetch(GHContent.class).wrap(this);
     }
 
+    /**
+     * Create repo variable
+     *
+     * @param name
+     *            the name
+     * @throws IOException
+     *             the io exception
+     */
+    public void createRepoVariable(final GHRepositoryVariable name) throws IOException {
+        root().createRequest()
+                .withUrlPath(getApiTailUrl("actions/variables"))
+                .method("POST")
+                .with("name", name.getName())
+                .with("value", name.getValue())
+                .send();
+    }
```

Comment:
Please move the implementation into a static method on `GHRepositoryVariable`. This method can remain here but the meat should be on the `GHRepositoryVariable` class.  See https://github.com/hub4j/github-api/blob/c357507750f9dd28b3732a2349ca8a1db4759e42/src/main/java/org/kohsuke/github/GHRepository.java#L2383C1-L2385 for an example.  

Also, probably rename to `createVariable` since the `Repo` is implied - if there are other `variable`s to be created on `GHRepository` I'm open to discussion here.  ... Ugh. I see the problem: https://github.com/hub4j/github-api/pull/1660 . You're quite reasonably following that naming convention. 😭  Hmm. 

```suggestion
    public void createVariable(final name, value) throws IOException {
      GHRepositoryVariable.create(this).name(name).value(value).done();
    }
```

The thing that particularly makes me think the `GHLabel` pattern is the way to go is your update method below that takes a separate `name` field to allow updating the name of an existing `GHRepositoryVariable`.

Reply:
Really inside GHRepository class, name a method with the word repository is redundant.